### DISCUSSION TITLE
CLI v2.0.0 Features Discussion

### DISCUSSION THREAD

**Discussion by Pradumnasaraf** [Pradumnasaraf/BioDrop-CLI#23] [Ideas]
Update the CLI with the latest features.

Features we need to work on:

| Feature | Description | Status |
| :--- | :--- | :--- |
| **BELOW LISTED FEATURES ARE FOR CREATE JSON FILE - MAIN PROMPT** |  |  |
| Add milestones | Add milestones to the profile | ✅ |
| Add tags | Add tags to the profile | ✅ |
| Add socials  | Add socials to the profile | ✅ |
| Add links | Add links to the profile | ✅ |
| **BELOW LISTED FEATURES ARE FOR UPDATE JSON FILE - MAIN PROMPT** |  |  |
| Update name | Update name to the profile | ✅ |
| Update bio | Update bio to the profile | ✅ |
| Update socials | Update socials to the profile | ✅ |
| Update links | Update links to the profile | ✅ |
| Update tags | Update tags to the profile | ✅|
| Update milestones | Update milestones to the profile | ✅ |
| Delete socials | Delete socials to the profile | ✅ |
| Delete links | Delete links to the profile | ✅ |
| Delete tags | Delete tags to the profile | ✅ |
| Delete milestones | Delete milestones to the profile | ✅ |
| **BELOW LISTED FEATURES ARE FOR GIVING TESTIMONAIALS TO A USER** |  |  |
| Add testimonials | Add testimonials to the profile | ✅ |
| **BELOW LISTED FEATURES ARE FOR ADDING A EVENT TO LINKFREE** |  |  |
| Add events | Add events to the profile | ✅|

**Comment by Pradumnasaraf** [comment]
I was thinking to start the prompt of CLI with something like the options:
```
Create a new profile
Update your profile
Add a testimonial.
Give a testimonial.
```
Also adding the #16 on @Ananya2001-an 's point.


_I will keep updating this message_

**Reply by Ananya2001-an** [reply]
> We should 1st update the CLI with the current features of LinkFree. 
> 
> Later, I was thinking to start the prompt of CLI with something like the options:
> ```
> Create a new profile
> Update your profile
> Give a testimonial.
> ```
> Also adding the #16 on @Ananya2001-an 's point.
> 

Sounds cool

**Reply by Pradumnasaraf** [reply]
Thank you. Let me know what your thought and views are and the points you want to put.

**Reply by Ananya2001-an** [reply]
> Thank you. Let me know what your thought and views are and the points you want to put.

Sure I will come back to you tomorrow regarding anything I wish to add

**Reply by Ananya2001-an** [reply]
> We should 1st update the CLI with the current features of LinkFree.
> 
> * Add tags
> * Add milestones
> 
> Later, I was thinking to start the prompt of CLI with something like the options:
> 
> ```
> Create a new profile
> Update your profile
> Add a testimonial.
> Give a testimonial.
> ```
> 
> Also adding the #16 on @Ananya2001-an 's point.
> 
> _I will keep updating this message_

We have to add socials and events as well.
Adding testimonials to your own profile can be part of the question function itself.
Whereas giving testimonial to someone else can be another option for the beginning so we can have 3 options:
Create a new profile
Update profile (in which we will have further choice to update a particular field)
Give testimonial for someone

What are your thoughts?

**Reply by Pradumnasaraf** [reply]
Hey,  Great points. Thank you. Just some follow-ups

> Adding testimonials to your profile can be part of the question function itself

By adding testimonials to the profile, I mean adding the person's GitHub username to our JSON file (Those who have given a testimonial to that person), so that it can reflect in their LinkFree profile. Something like this. https://linkfree.eddiehub.io/docs/how-to-guides/testimonials

- How do you think we should do the update profile thing? Because data is in the LinkFree repo.

POST:
> Because data is in the LinkFree repo

What a silly comment I put, the CLI is used in LinkFree itself, so it will have access to the file. 🤦‍♂️

**Reply by Ananya2001-an** [reply]
Okay 👍 got it about the testimonials. For updating profile can't we just read the file content using fs and then parse it using JSON.parse to fetch the key value pairs. 

fs.readFile(filePath, function (error, content) {
    var data = JSON.parse(content);
});

**Reply by Pradumnasaraf** [reply]
Yes, correct. But my concern is in JSON files we don't have unique keys-value in child elements (inside links and socials). Like links is an array. So how can we convert those in form of a CLI prompt? I am just a little concerned about the complexity, WDYT? I hope you have a better approach.

Array i mean this

<img width="517" alt="Screenshot 2023-02-16 at 11 27 01 AM" src="https://user-images.githubusercontent.com/51878265/219280926-53e3126c-4bfc-45ad-bb74-a6445d82e4e2.png">

**Reply by Ananya2001-an** [reply]
While updating profile when we come to links we can ask user for 3 choices :
add a link
edit a link
remove a link
then depending on the choice we can go for the process. If user decides to remove or edit a link we can then traverse through each element in array and show the url to the user so that they know which link it is. If they select it then we call a function for the operation to be performed. I guess that is how we can deal with this. I know it's complex but there are things that we need to take care of. We will have to do a lot of refactoring. Then same can be done for other keys with array values. What do u say??

**Reply by Pradumnasaraf** [reply]
Sounds great. Got your point. This sounds much better. We can do something similar like an icon selection approach (prompt process).

https://github.com/Pradumnasaraf/LinkFree-CLI/blob/d255814896c04477e0158a790f9ceb7803d46497/src/util/questions.js#L33

<img width="718" alt="Screenshot 2023-02-16 at 12 02 06 PM" src="https://user-images.githubusercontent.com/51878265/219286716-7a32204f-7bf2-4abc-82bb-ff66e5c2d316.png">

**Reply by Ananya2001-an** [reply]
Yeah that will work.

**Reply by Pradumnasaraf** [reply]
Great. 

I think we should clean the index file by separating the features into different folders. So that every feature can be developed independently. Like createuser, update, etc

**Reply by Ananya2001-an** [reply]
Yes we should do that

**Reply by Pradumnasaraf** [reply]
Awesome. I will go ahead and try to do a little clean-up and will push on the `refactor-cli` branch https://github.com/Pradumnasaraf/LinkFree-CLI/tree/refactor-cli.

**Reply by Ananya2001-an** [reply]
Great I will review the changes side by side.

**Comment by Pradumnasaraf** [comment]
Hey, @Ananya2001-an

I have pushed the changes to the https://github.com/Pradumnasaraf/LinkFree-CLI/tree/refactor-cli. I have broken the functionality into folders. You can have a look. Feel free to do any changes/refactoring you want. Also, I have added a `Notes.md` for any kind of message we want to convey.

**Reply by Ananya2001-an** [reply]
Cool. I will look into it.

**Reply by Ananya2001-an** [reply]
Hey @Pradumnasaraf I have done some changes and I will commit them soon so that you can see them as well.

**Reply by Pradumnasaraf** [reply]
> Hey @Pradumnasaraf I have done some changes and I will commit them soon so that you can see them as well. 

Awesome.

**Reply by Ananya2001-an** [reply]
Hey @Pradumnasaraf just raised a PR right now. Do check the changes and tell me your views on it.

**Reply by Pradumnasaraf** [reply]
Hey,
Sure I will look at the PR. It is a big PR with 300+ lines of change. 

Interestingly I also work on milestones and social features. No worries I will prioritize your changes and go with them.

**Reply by Ananya2001-an** [reply]
> Hey,
> Sure I will look at the PR. It is a big PR with 300+ lines of change. 
> 
> Interestingly I also work on milestones and social features. No worries I will prioritize your changes and go with them.

You can definitely tweak the code the way you want.

**Reply by Pradumnasaraf** [reply]
Thank you. 

Instead, I will leave comments and reviews on the PR itself. That will be more transparent and easy to brainstorm

**Comment by Pradumnasaraf** [comment]
Hey, @Ananya2001-an, Thanks for PR again. How do you think we should proceed next? 

I am thinking to go ahead and add the tags and testimonials (main prompt) feature and then raise a PR.

**Reply by Ananya2001-an** [reply]
> Hey, @Ananya2001-an, Thanks for PR again. How do you think we should proceed next? 
> 
> I am thinking to go ahead and add the tags and testimonials (main prompt) feature and then raise a PR.

Yeah right now for createJson we need to add tags and socials. And later you can work on the events and testimonials because they will have to be in a separate folder.

**Reply by Pradumnasaraf** [reply]
Great point. Sound good. Should I go ahead with adding tags and add `tags` and `socials`?

**Reply by Ananya2001-an** [reply]
> Great point. Sound good. Should I go ahead with adding tags and add `tags` and `socials`?

Yes that would be good.

**Reply by Pradumnasaraf** [reply]
Hey, I just saw I did a silly mistake, I said milestones instead of testimonials (but below I added an explanation for testimonials ) 🤦 . I am so sorry for that.  This is why you commented that out.

<img width="766" alt="Screenshot 2023-02-17 at 4 58 04 PM" src="https://user-images.githubusercontent.com/51878265/219640417-0ffa38d4-8ac5-472c-b7c6-beed54152788.png">

> We should add testimonials as a separate thing.

I will fix that also. Again sorry.

**Reply by Ananya2001-an** [reply]
> Hey, I just saw I did a silly mistake, I said milestones instead of testimonials (but below I added an explanation for testimonials ) 🤦 . I am so sorry for that.  This is why you commented that out.
> 
> <img width="766" alt="Screenshot 2023-02-17 at 4 58 04 PM" src="https://user-images.githubusercontent.com/51878265/219640417-0ffa38d4-8ac5-472c-b7c6-beed54152788.png">
> 
> I will fix that also. Again sorry.
> 

Yeah I actually thought that maybe you are talking about testimonials. No worries if you want you can uncomment the milestones prompt that I created and add it for createJson later.

**Reply by Pradumnasaraf** [reply]
Ahh 😅 

I think uncommenting will won't work because it's not imported and its functionality is also commented in `basic.js`, so I think I should create a separate file in a shared one so that we can also use it in `update JSON` also.

**Reply by Ananya2001-an** [reply]
> Ahh 😅 
> 
> I think uncommenting will won't work because it's not imported and its functionality is also commented in `basic.js`, so I think I should create a separate file in a shared one so that we can also use it in `update JSON` also.

Yeah that would be be great

**Reply by Pradumnasaraf** [reply]
Thank you for fast response and time

**Reply by Ananya2001-an** [reply]
Well I use github on my phone to reply to any messages so it's almost like using whatsapp for me. 😅

**Reply by Pradumnasaraf** [reply]
> Well I use github on my phone to reply to any messages so it's almost like using whatsapp for me. 😅

Likewise 😬

**Comment by Pradumnasaraf** [comment]
Hey @Ananya2001-an,

What do you think about how we should move forward? Also, before that, It's my responsibility to ask if you are still comfortable with further development together.

Feel free to reply after the weekend. Happy weekend!

**Reply by Ananya2001-an** [reply]
> Hey @Ananya2001-an,
> 
> What do you think about how we should move forward? Also, before that, It's my responsibility to ask if you are still comfortable with further development together.
> 
> Feel free to reply after the weekend. Happy weekend!
> 

Of course I am still willing to work together. 😊 We can maybe focus now on completing those questions for delete and update as well. It will be like copy pasting only. Once that is done then we can work on the events and testimonials (main prompt) which will be in separate folder.

**Reply by Ananya2001-an** [reply]
Also congrats on your CFP for kubehuddle.👏 Just saw your tweet today on Twitter.

**Reply by Pradumnasaraf** [reply]
> Of course I am still willing to work together. 😊 We can maybe focus now on completing those questions for delete and update as well. It will be like copy pasting only. Once that is done then we can work on the events and testimonials (main prompt) which will be in separate folder.

Awesome. Would to like to work on the delete and update part?

Again, thank you for your time.

**Reply by Pradumnasaraf** [reply]
> Also congrats on your CFP for kubehuddle.👏 Just saw your tweet today on Twitter.

Thank you, Let's see if they will accept or not.

**Reply by Pradumnasaraf** [reply]
Also, a huge shoutout to you for coming up with wonderful dir structuring (`shared`  dir one). Now, adding something has become so easy. Also, now adding things doesn't disturb other pieces of the app.

**Reply by Ananya2001-an** [reply]
> Also, a huge shoutout to you for coming up with wonderful dir structuring (`shared`  dir one). Now, adding something has become so easy. Also, now adding things doesn't disturb other pieces of the app.

Thanks pradumna. Web dev tutorials have finally proved their worth today for me😂

**Reply by Ananya2001-an** [reply]
> > Of course I am still willing to work together. 😊 We can maybe focus now on completing those questions for delete and update as well. It will be like copy pasting only. Once that is done then we can work on the events and testimonials (main prompt) which will be in separate folder.
> 
> Awesome. Would to like to work on the delete and update part?
> 
> Again, thank you for your time.
> 

Yeah I can

**Reply by Ananya2001-an** [reply]
> > Also congrats on your CFP for kubehuddle.👏 Just saw your tweet today on Twitter.
> 
> Thank you, Let's see if they will accept or not.

Don't worry they will. Hope I can submit my own someday. I am still learning.

**Reply by Pradumnasaraf** [reply]
> Don't worry they will. Hope I can submit my own someday. I am still learning.

Don't think too much. You already know JS and Node. There are lots of conferences just looking for that. At most what will happen, they will say NO.

**Reply by Ananya2001-an** [reply]
> > Don't worry they will. Hope I can submit my own someday. I am still learning.
> 
> Don't think too much. You already know JS and Node. There are lots of conferences just looking for that. At most what will happen, they will say NO.

My demotivated brain cells just got the right amount of serotonin 😂 thnx for the kind words. I will definitely try.

**Comment by Pradumnasaraf** [comment]
This message for discussion is to keep track of minor things we need to fix. Either these things need to explain well or some functionality needs to be changed. THESE ARE NOT A BLOCKER FOR RIGHT NOW. Later we can raise an issue for each of them and fix them.

1) **If a person doesn't update the profile. The final message should be something like - "Nothing got updated" instead successful message.**

<img width="666" alt="Screenshot 2023-02-18 at 5 53 11 PM" src="https://user-images.githubusercontent.com/51878265/219865600-c46ce87b-e156-4742-bc0d-f96ed6643810.png">

2) **In the update profile, we need to add a little clear message about which testimonials it's updating, the usernames, or the person given**

<img width="429" alt="Screenshot 2023-02-18 at 5 57 21 PM" src="https://user-images.githubusercontent.com/51878265/219865739-cd7fe0ec-3e09-4959-a6a0-86d5de8fad06.png">

3) **When we delete all the elements by the update option. It should remove the empty key**

4) **There are some keys that are optional and but we ask the user to enter the value. For eg, `color` and `links` in the milestone. So either we need to ask if they wan't to add them (choice) or remove the empty keys somehow at the end before writing to the file. Hope you got what I am trying to explain.**

 <img width="486" alt="Screenshot 2023-02-20 at 7 50 06 PM" src="https://user-images.githubusercontent.com/51878265/220132697-d91e3b71-e190-414c-8053-b8401d233c2b.png">

**Reply by Ananya2001-an** [reply]
> This message for discussion is to keep track of minor things we need to fix. Either these things need to explain well or some functionality needs to be changed. THESE ARE NOT A BLOCKER FOR RIGHT NOW. Later we can raise an issue for each of them and fix them.
> 
> 1) **If a person doesn't update the profile. The final message should be something like - "Nothing got updated" instead successful message.**
> 
> <img width="666" alt="Screenshot 2023-02-18 at 5 53 11 PM" src="https://user-images.githubusercontent.com/51878265/219865600-c46ce87b-e156-4742-bc0d-f96ed6643810.png">
> 
> 2) **In the update profile, we need to add a little clear message about which testimonials it's updating, the usernames, or the person given**
> 
> <img width="429" alt="Screenshot 2023-02-18 at 5 57 21 PM" src="https://user-images.githubusercontent.com/51878265/219865739-cd7fe0ec-3e09-4959-a6a0-86d5de8fad06.png">
> 
> 

Regarding the second point if we said that we are updating the JSON profile that means it's the usernames and not their own testimonial that they gave to someone else, right?

**Reply by Pradumnasaraf** [reply]
I agree 100%. This makes very less sense. But It's written "Your", and you know some people can easily misinterpret that.

We can just add a username in the bracket. Like this. What do you think?
```
Do you want to update your testimonials (usernames)?
```

**Reply by Ananya2001-an** [reply]
> I agree 100%. This makes very less sense. But It's written "Your", and you know some people can easily misinterpret that.
> 
> We can just add a username in the bracket. Like this. What do you think?
> ```
> Do you want to update your testimonials (usernames)?
> ```

Yeah that looks good

**Comment by Ananya2001-an** [comment]
> Update the CLI with the latest features.
> 
> Features we need to work on:
> 
> | Feature | Description | Status |
> | :--- | :--- | :--- |
> | **BELOW LISTED FEATURES ARE FOR CREATE JSON FILE - MAIN PROMPT** |  |  |
> | Add milestones | Add milestones to the profile | ✅ |
> | Add tags | Add tags to the profile | ✅ |
> | Add socials  | Add socials to the profile | ✅ |
> | Add links | Add links to the profile | ✅ |
> | **BELOW LISTED FEATURES ARE FOR UPDATE JSON FILE - MAIN PROMPT** |  |  |
> | Update name | Update name to the profile | ✅ |
> | Update bio | Update bio to the profile | ✅ |
> | Update socials | Update socials to the profile | ❌ |
> | Update links | Update links to the profile | ✅ |
> | Update tags | Update tags to the profile | ❌ |
> | Update milestones | Update milestones to the profile | ❌ |
> | Delete socials | Delete socials to the profile | ❌ |
> | Delete links | Delete links to the profile | ✅ |
> | Delete tags | Delete tags to the profile | ❌ |
> | Delete milestones | Delete milestones to the profile | ❌ |
> | **BELOW LISTED FEATURES ARE FOR GIVING TESTIMONAIALS TO A USER** |  |  |
> | Add testimonials | Add testimonials to the profile | ❌ |
> | **BELOW LISTED FEATURES ARE FOR ADDING A EVENT TO LINKFREE** |  |  |
> | Add events | Add events to the profile | ❌ |
> 

This looks great 👍

**Reply by Pradumnasaraf** [reply]
Thank you, Ananya

It will give a little more clarity and help in tracking.

**Comment by Pradumnasaraf** [comment]
Hey, @Ananya2001-an

Again thanks for the PR. What should we do next?

**Reply by Ananya2001-an** [reply]
> Hey, @Ananya2001-an
> 
> Again thanks for the PR. What should we do next? 

You can't keep on thanking me all the time 😂. Well I guess you can now work on the events and testimonials stuff if that's OK for you? And after that I think we can look after the small fixes that are needed to be made. I guess then it will be finally over.

**Reply by Pradumnasaraf** [reply]
> You can't keep on thanking me all the time 😂.

Yes, because in today's world people don't appreciate others. A small sentence of kind words can motivate and boosts morale :)

**Reply by Pradumnasaraf** [reply]
> Well I guess you can now work on the events and testimonials stuff if that's OK for you? And after that I think we can look after the small fixes that are needed to be made. I guess then it will be finally over.

Sounds cool. I will go ahead with testimonials. I have something in mind. Thanks

**Reply by Ananya2001-an** [reply]
> > Well I guess you can now work on the events and testimonials stuff if that's OK for you? And after that I think we can look after the small fixes that are needed to be made. I guess then it will be finally over.
> 
> Sounds cool. I will go ahead with testimonials. I have something in mind. Thanks

Great👍

**Reply by Pradumnasaraf** [reply]
Also, I have raised a Draft PR for `v2.0.0` for visibility https://github.com/Pradumnasaraf/LinkFree-CLI/pull/28 .

**Reply by Ananya2001-an** [reply]
> Also, I have raised a Draft PR for `v2.0.0` for visibility https://github.com/Pradumnasaraf/LinkFree-CLI/pull/28 .

Awesome 👌

**Reply by Pradumnasaraf** [reply]
Hey,

Now need to work on the final major part, **Adding events**. Would you like to work on this or I should do that?

**Reply by Ananya2001-an** [reply]
> Hey,
> 
> Now need to work on the final major part, **Adding events**. Would you like to work on this if I should do that?

I can try but if you want to do it then it's your choice.

**Reply by Pradumnasaraf** [reply]
Ah, this reply confused me 😅. Not sure what to say next

**Reply by Ananya2001-an** [reply]
> Ah, this reply confused me 😅. Not sure what to say next

Sorry to confuse you. I just thought that maybe you already have an idea on how to work on this that's why I said that it's your call if you feel like doing it then you can otherwise I will do it.

**Reply by Pradumnasaraf** [reply]
Nothing to be sorry about. 

No, currently I don't have any idea. But I think it will be quick to build because it will be on those same lines as the milestone. If you like you can go ahead, no pressure.

I was just thinking about how we should deal with the `date` thing. Because that is the tricky part and room for lots of error. I was just thinking can we take in a simpler form and convert it internally, not sure.

This is the data we need to collect.
```json
{
  "userStatus": "speaking",
  "speakerDetails": "Cloud Computing with Spring Cloud",
  "isVirtual": true,
  "isInPerson": true,
  "color": "red",
  "name": "Open Source GitHub reviews",
  "description": "In this livestream I will be going reviewing your **Open Source projects** and profiles! I will be joined by **Amanda**, a Developer Advocate.",
  "date": { // this part
    "start": "2022-12-09T16:00:00.000+00:00",
    "end": "2022-12-09T17:00:00.000+00:00",
    "cfpClose": "2022-10-09T17:00:00.000+00:00"
  },
  "url": "https://www.youtube.com/watch?v=iqIFD02OkVE"
}
```

**Reply by Ananya2001-an** [reply]
Can't we take the date from user as string and then convert it into a Date object which will give us this format I guess?

**Reply by Ananya2001-an** [reply]
> Nothing to be sorry about. 
> 
> No, currently I don't have any idea. But I think it will be quick to build because it will be on those same lines as the milestone. If you like you can go ahead, no pressure.
> 
> I was just thinking about how we should deal with the `date` thing. Because that is the tricky part and room for lots of error. I was just thinking can we take in a simpler form and convert it internally, not sure.
> 
> This is the data we need to collect.
> ```json
> {
>   "userStatus": "speaking",
>   "speakerDetails": "Cloud Computing with Spring Cloud",
>   "isVirtual": true,
>   "isInPerson": true,
>   "color": "red",
>   "name": "Open Source GitHub reviews",
>   "description": "In this livestream I will be going reviewing your **Open Source projects** and profiles! I will be joined by **Amanda**, a Developer Advocate.",
>   "date": { // this part
>     "start": "2022-12-09T16:00:00.000+00:00",
>     "end": "2022-12-09T17:00:00.000+00:00",
>     "cfpClose": "2022-10-09T17:00:00.000+00:00"
>   },
>   "url": "https://www.youtube.com/watch?v=iqIFD02OkVE"
> }
> ```
> 
> 
> 

Sure I can work on it no issues.

**Reply by Pradumnasaraf** [reply]
Yes, I was thinking of something similar. 

Also, I just checked the `userStatus` will in the form options. Docs for ref - https://linkfree.eddiehub.io/docs/how-to-guides/events
![Screenshot 2023-02-20 at 6 50 44 PM](https://user-images.githubusercontent.com/51878265/220120234-095df93c-99e2-481f-9359-46a6a5ef828c.png)

**Reply by Ananya2001-an** [reply]
> Yes, I was thinking of something similar. 
> 
> Also, I just checked the `userStatus` will in the form options. Docs for ref - https://linkfree.eddiehub.io/docs/how-to-guides/events
> ![Screenshot 2023-02-20 at 6 50 44 PM](https://user-images.githubusercontent.com/51878265/220120234-095df93c-99e2-481f-9359-46a6a5ef828c.png)
> 

Means we have to create a select prompt for it given these 2 choices?

**Reply by Ananya2001-an** [reply]
Actually I had this doubt that like userstatus there are many keys which are not necessary to fill but if the user wants to add them then what we will do? Like adding group for link key. It's not important for now but we will have to look after it later.

**Reply by Pradumnasaraf** [reply]
> Means we have to create a select prompt for it given these 2 choices?

Yes

> Actually I had this doubt that like userstatus there are many keys which are not necessary to fill but if the user wants to add them then what we will do? Like adding group for link key. It's not important for now but we will have to look after it later.

Yes, I want to discuss this thing with you. But I thought to discuss it later because things can be complex the development. I have some ideas. Let's do the major things first and brainstorm together later.

**Reply by Ananya2001-an** [reply]
> > Means we have to create a select prompt for it given these 2 choices?
> 
> Yes
> 
> > Actually I had this doubt that like userstatus there are many keys which are not necessary to fill but if the user wants to add them then what we will do? Like adding group for link key. It's not important for now but we will have to look after it later.
> 
> Yes, I want to discuss this thing with you. But I thought to discuss it later because things can be complex the development. I have some ideas. Let's do the major things first and brainstorm together later.
> 
> 

Okay I will start with the events soon then.👍

**Reply by Pradumnasaraf** [reply]
Also I want to dicuss one more thing I totally forgot about that.

There are some keys which are optional and but we ask user to enter the value. For eg, `color` and `links` in the milestone. So either we need to ask if they wan't to add them (choise) or remove the empty keys somehow at the end before writing to the file. Hope you got what I am trying to explain.

Screenshot for better clarity

<img width="486" alt="Screenshot 2023-02-20 at 7 50 06 PM" src="https://user-images.githubusercontent.com/51878265/220132697-d91e3b71-e190-414c-8053-b8401d233c2b.png">

**Not a blocker right now**

**Reply by Ananya2001-an** [reply]
> Also I want to dicuss one more thing I totally forgot about that.
> 
> There are some keys which are optional and but we ask user to enter the value. For eg, `color` and `links` in the milestone. So either we need to ask if they wan't to add them (choise) or remove the empty keys somehow at the end before writing to the file. Hope you got what I am trying to explain.
> 
> Screenshot for better clarity
> 
> <img width="486" alt="Screenshot 2023-02-20 at 7 50 06 PM" src="https://user-images.githubusercontent.com/51878265/220132697-d91e3b71-e190-414c-8053-b8401d233c2b.png">
> 
> **Not a blocker right now**
> 
> 

Yeah that's basically what we need to do for all such optional keys.

**Reply by Pradumnasaraf** [reply]
> Okay I will start with the events soon then.👍

No rush. If you need help, tell me. If you want I can help in writing prompts

**Reply by Ananya2001-an** [reply]
> > Okay I will start with the events soon then.👍
> 
> No rush. If you need help, tell me. If you want I can help in writing prompts

No worries I guess I can do it. If I need your help I will definitely bug you 🐛

**Reply by Pradumnasaraf** [reply]
> > > Okay I will start with the events soon then.👍
> > 
> > 
> > No rush. If you need help, tell me. If you want I can help in writing prompts
> 
> No worries I guess I can do it. If I need your help I will definitely bug you 🐛

Awesome. Thank you.
